Panic on load from immutable DB

I ran into this on a local dingo run in preview when the data dir was still populated from a previous run, and the local tip was later than the immutable DB tip.

```
{"time":"2025-09-11T20:02:09.072050665Z","level":"INFO","msg":"copying blocks from immutable DB"}
panic: runtime error: index out of range [18193] with length 18193

goroutine 1 [running]:
github.com/blinklabs-io/dingo/database/immutable.(*ImmutableDb).getChunkNamesFromPoint(0xc0068aa300, {{}, 0x0?, {0xc00681f100?, 0x0?, 0x56c0356?}})
        /home/agaffney/dev/blinklabs-io/dingo/database/immutable/immutable.go:85 +0x20e
github.com/blinklabs-io/dingo/database/immutable.(*ImmutableDb).BlocksFromPoint(0xc0068aa300, {{}, 0x241d688?, {0xc00681f100?, 0x0?, 0x20317fa?}})
        /home/agaffney/dev/blinklabs-io/dingo/database/immutable/immutable.go:274 +0x38
```

The failing line is here:

https://github.com/blinklabs-io/dingo/blob/2ab6fbba0e589431d35196493f745d581e2ae102/database/immutable/immutable.go#L85

--- a/database/immutable/immutable.go
+++ b/database/immutable/immutable.go
@@ -82,6 +82,13 @@ func (i *ImmutableDb) getChunkNamesFromPoint(
 	for lowerBound <= upperBound {
 		// Get chunk in the middle of the current bounds
 		middlePoint := (lowerBound + upperBound) / 2
+		if middlePoint < 0 || middlePoint >= len(chunkNames) {
+			return nil, fmt.Errorf(
+				"immutable DB: middlePoint %d out of bounds (chunkNames len=%d)",
+				middlePoint,
+				len(chunkNames),
+			)
+		}
 		middleChunkName := chunkNames[middlePoint]
 		middleSecondary, err := i.getChunkSecondaryIndex(middleChunkName)
 		if err != nil {

